flows/boundary.py

```diff
@@ -545,6 +547,112 @@ def get_document_passage_from_vespa(
+    structure that exist for each of the hits in the group.
+    """
+
+    continuation_tokens = []
+
+    vespa_query_response_root = vespa_query_response.json["root"]
+    group_hits = dig(vespa_query_response_root, "children", 0, "children", default=[])
+    for hit in group_hits:
+        hit_continuation_token = dig(hit, "continuation", "next", default=None)
+        if hit_continuation_token:
+            continuation_tokens.append(hit_continuation_token)
+    return continuation_tokens
+
+
+def get_vespa_passages_from_query_response(
+    vespa_query_response: VespaQueryResponse,
+) -> list[tuple[str, VespaPassage]]:
+    """Retrieve the passages from the query response."""
+
+    vespa_query_response_root = vespa_query_response.json["root"]
+    passages_root = dig(
+        vespa_query_response_root, "children", 0, "children", 0, "children", default=[]
+    )
+    passage_hits = [
+        dig(passage_root, "children", 0, "children", 0)
+        for passage_root in passages_root
+    ]
+    vespa_passages: list[tuple[str, VespaPassage]] = [
+        (passage["id"], VespaPassage.model_validate(passage["fields"]))
+        for passage in passage_hits
+    ]
+
+    return vespa_passages
+
+
+def get_document_passages_from_vespa__generator(
+    document_import_id: DocumentImportId,
+    vespa_search_adapter: VespaSearchAdapter,
+    continuation_tokens: list[str],
+    grouping_max: int = 10,
+) -> Generator[list[tuple[str, VespaPassage]], None, None]:
+    conditions = qb.QueryField("family_document_ref").contains(
+        f"id:doc_search:family_document::{document_import_id}"
+    )
+
+    grouping = G.all(
+        G.group("text_block_id"),
+        G.max(grouping_max),
+        G.each(G.each(G.output(G.summary()))),
+    )
+
+    while continuation_tokens:
+        query: qb.Query = (
+            qb.select("*")  # type: ignore
+            .from_(
+                Schema(name="document_passage", document=Document()),
+            )
+            .where(conditions)
+            .set_limit(0)
+            .groupby(grouping, continuations=continuation_tokens)
+        )
+
+        vespa_query_response: VespaQueryResponse = vespa_search_adapter.client.query(
+            yql=query
+        )
+
+        if not vespa_query_response.is_successful():
+            raise QueryError(vespa_query_response.get_status_code())
+
+        # TODO: Do we need to handle empty responses here like:
+        # {
+        #     "root": {
+        #         "id": "toplevel",
+        #         "relevance": 1.0,
+        #         "fields": {
+        #             "totalCount": 0
+        #         },
+        #         "coverage": {
+        #             "coverage": 100,
+        #             "documents": 0,
+        #             "full": true,
+        #             "nodes": 1,
+        #             "results": 1,
+        #             "resultsFull": 1
+        #         }
+        #     }
+        # }
```

Comment:
In this instance we would yield no passages due to the conditional below so maybe we're fine.